services/src/handlers/model_training.rs

```diff
@@ -0,0 +1,199 @@
+use std::sync::Arc;
+
+use actix_web::{web, Responder};
+use futures::stream::select_all;
+use futures::{FutureExt, StreamExt, TryFutureExt, TryStreamExt};
+use geoengine_datatypes::primitives::VectorQueryRectangle;
+
+use geoengine_datatypes::raster::{ConvertDataTypeParallel, GridOrEmpty};
+use geoengine_operators::call_on_generic_raster_processor;
+use geoengine_operators::engine::{
+    ExecutionContext, InitializedRasterOperator, QueryContext, QueryProcessor, RasterOperator,
+    TypedRasterQueryProcessor,
+};
+use geoengine_operators::util::spawn_blocking_with_thread_pool;
+
+use super::tasks::TaskResponse;
+use crate::contexts::ApplicationContext;
+use crate::error::Error;
+use crate::error::Result;
+use crate::machine_learning::{
+    schedule_ml_model_from_workflow_task, xg_error::XGBoostModuleError as XgModuleError,
+    Aggregatable, MLTrainRequest, MachineLearningAggregator, MachineLearningFeature,
+    ReservoirSamplingAggregator, SimpleAggregator,
+};
+use crate::workflows::workflow::Workflow;
+
+#[utoipa::path(
+    tag = "Machine Learning",
+    post,
+    path = "/ml/train",
+    request_body = MLTrainRequest,
+    responses(
+        (
+            status = 200, description = "Model training from workflows", body = TaskResponse,
+            example = json!({"taskId": "7f8a4cfe-76ab-4972-b347-b197e5ef0f3c"})
+        )
+    ),
+    security(
+        ("session_token" = [])
+    )
+)]
+pub async fn ml_model_from_workflow_handler<C: ApplicationContext>(
+    session: C::Session,
+    app_ctx: web::Data<C>,
+    info: web::Json<MLTrainRequest>,
+) -> Result<impl Responder> {
+    let ctx = Arc::new(app_ctx.session_context(session));
+
+    let task_id = schedule_ml_model_from_workflow_task(
+        info.input_workflows.clone(),
+        info.label_workflow.clone(),
+        ctx,
+        info.into_inner(),
+    )
+    .await?;
+
+    Ok(web::Json(TaskResponse::new(task_id)))
+}
+
+/// This method initializes the raster operators and produces a vector of typed raster query
+/// processors.
+pub async fn get_query_processors(
```

Comment:
move these methods to the machine_learning module and do not import stuff from the handler there.